synapse/handlers/e2e_room_keys.py

```diff
@@ -290,8 +302,45 @@ def delete_version(self, user_id, version=None):
             user_id(str): the user whose current backup version we're deleting
             version(str): the version id of the backup being deleted
         Raises:
-            StoreError: code 404 if this backup version doesn't exist
+            NotFoundError: if this backup version doesn't exist
         """
 
         with (yield self._upload_linearizer.queue(user_id)):
-            yield self.store.delete_e2e_room_keys_version(user_id, version)
+            try:
+                yield self.store.delete_e2e_room_keys_version(user_id, version)
+            except StoreError as e:
+                if e.code == 404:
+                    raise NotFoundError("Unknown backup version")
+                else:
+                    raise
+
+    @defer.inlineCallbacks
+    def update_version(self, user_id, version, version_info):
+        """Update the info about a given version of the user's backup
+
+        Args:
+            user_id(str): the user whose current backup version we're updating
+            version(str): the backup version we're updating
+            version_info(dict): the new information about the backup
+        Raises:
+            NotFoundError: if the requested backup version doesn't exist
+        Returns:
+            A deferred of an empty dict.
+        """
+        try:
```

Comment:
Other functions in this file are wrapped in `with (yield self._upload_linearizer.queue(user_id)):`.  I don't know if this function needs it too.

Reply:
I would probably do so, it may not be necessary but the fact that we do it elsewhere and we do two separate database queries makes me inclined to add it for consistency.

(FTR, all the function does is make sure only one set of code is being executed for a user at any given point)